Remotion/ObjectBinding/Sample/InMemoryReflectionBusinessObjectStorageProvider.cs

```diff
@@ -0,0 +1,77 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Remotion.Utilities;
+
+namespace Remotion.ObjectBinding.Sample
+{
+  public class InMemoryReflectionBusinessObjectStorageProvider : IReflectionBusinessObjectStorageProvider
+  {
+    private readonly Dictionary<Tuple<Type, Guid>, MemoryStream> _reflectionBusinessObjectData = new Dictionary<Tuple<Type, Guid>, MemoryStream>();
```

Comment:
Should we use a ValueTuple to make it better readable (i.e. use named items in the tuple)? Or should we use a nested dictionary?

Actually, we have an AutoInitDictionary, so we could go:
`AutoInitDictionary<Type,Dictionary<Guid,MemoryStream>>`

Accessing would then be `dict[type].TryGetValue(guid, out memoryStream)` 
or `dict[type].GetValueOrDefault(guid)`  (we have an extension method for this)